src/components/ConnectivityChecker.vue

```diff
@@ -29,15 +32,40 @@ export default {
       },
       false
     );
+    window.addEventListener(
+      "online",
+      function () {
+          that.checkOffline();
+      },
+      false
+    );
+    window.addEventListener(
+      "offline",
+      function () {
+        this.offline = true;
+      },
+      false
+    );
   },
   methods: {
     checkOffline: function () {
+      if (!navigator.onLine) {
+        this.offline = true;
+        return;
+      }
+
+      // extra check to make sure we're not offline
       let that = this;
       return fetch(window.location.href + "?alive", {
         method: "HEAD",
         cache: "no-store",
+        redirect: "manual"
       })
         .then(function (response) {
+          // opaqueredirect means request has been redirected, to auth provider probably
+          if ((response.type === "opaqueredirect" && !response.ok) || [401, 403].indexOf(response.status) != -1) {
+            window.location.href = window.location.href + "?t="+(new Date().valueOf());
```

Comment:
I think we should use the same parameter everywhere, and the timestamp value as well, you should replace the ?alive parameter in the original request by the `t=<timestamp>`

Reply:
I also think that if we can prevent the `alive` to be cached, we should do it.